src/remote.rs

```diff
@@ -43,13 +44,17 @@ pub struct FetchOptions<'cb> {
     prune: FetchPrune,
     update_fetchhead: bool,
     download_tags: AutotagOption,
+    custom_headers: Vec<CString>,
+    custom_headers_ptrs: Vec<*const c_char>,
```

Comment:
why do we need the duplication?
also I think the best practice is to keep stuff in `Option` by default.

Reply:
Nah I think this is ok, empty vecs don't need allocation and one here is the owned vector of strings and the other is what we pass to libgit2 itself